src/main/java/com/google/genai/BraintrustApiClient.java

```diff
@@ -0,0 +1,416 @@
+package com.google.genai;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.genai.types.HttpOptions;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nullable;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Headers;
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+
+/**
+ * Instrumented wrapper for ApiClient that adds OpenTelemetry spans.
+ *
+ * <p>This class lives in com.google.genai package to access package-private ApiClient class.
+ */
+@Slf4j
+class BraintrustApiClient extends ApiClient {
+    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+    private final ApiClient delegate;
+    private final Tracer tracer;
+
+    public BraintrustApiClient(ApiClient delegate, OpenTelemetry openTelemetry) {
+        // We must call super(), but we'll override all methods to delegate
+        // Pass the delegate's config to minimize differences
+        super(
+                delegate.apiKey != null ? delegate.apiKey : Optional.empty(),
+                delegate.project != null ? delegate.project : Optional.empty(),
+                delegate.location != null ? delegate.location : Optional.empty(),
+                delegate.credentials != null ? delegate.credentials : Optional.empty(),
+                delegate.httpOptions != null ? Optional.of(delegate.httpOptions) : Optional.empty(),
+                delegate.clientOptions != null ? delegate.clientOptions : Optional.empty());
+        this.delegate = delegate;
+        this.tracer = openTelemetry.getTracer("io.opentelemetry.gemini-java-1.20");
+    }
+
+    private void tagSpan(
+            Span span,
+            @Nullable String genAIEndpoint,
+            @Nullable String requestMethod,
+            @Nullable String requestBody,
+            @Nullable String responseBody) {
+        try {
+            Map<String, Object> metadata = new java.util.HashMap<>();
+            metadata.put("provider", "gemini");
+
+            // Parse request
+            if (requestBody != null) {
+                var requestJson = JSON_MAPPER.readValue(requestBody, Map.class);
+
+                // Extract metadata fields
+                for (String field :
```

Comment:
my recommendation is to pass the literal input and output to the span. at rest it'll look liek 

<img width="724" height="1247" alt="Image" src="https://github.com/user-attachments/assets/a802f896-518e-41c4-9628-1f63e73f1cd6" />

([ref](https://www.braintrust.dev/app/braintrustdata.com/p/golden-ts-genai/logs?range=%22all%22&r=08bba73c-da9f-4889-b7e7-ea3cb3a83389&s=7766ef6d-2e39-4263-9dc5-605f80d4ade8))

you may find that the ui will automatically render java produced spans automatically